clients/src/main/java/org/apache/kafka/common/network/SizeDelimitedSend.java

```diff
@@ -14,18 +14,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-package org.apache.kafka.common.protocol;
+package org.apache.kafka.common.network;
 
 import java.nio.ByteBuffer;
 
-public final class MessageTestUtil {
-    public static ByteBuffer messageToByteBuffer(Message message, short version) {
-        ObjectSerializationCache cache = new ObjectSerializationCache();
-        int size = message.size(cache, version);
-        ByteBuffer bytes = ByteBuffer.allocate(size);
-        message.write(new ByteBufferAccessor(bytes), cache, version);
-        bytes.rewind();
-        return bytes;
+/**
+ * A size delimited Send that consists of a 4 byte network-ordered size N followed by N bytes of content.
+ */
+public class SizeDelimitedSend extends ByteBufferSend {
```

Comment:
How about making ```SizeDelimitedSend``` be a static method in ```ByteBufferSend```? For example:
```java
    public static Send withSizeDelimited(ByteBuffer buffer) {
        ByteBuffer sizeBuffer = ByteBuffer.allocate(4);
        sizeBuffer.putInt(0, buffer.remaining());
        return new ByteBufferSend(sizeBuffer, buffer);
    }
```

Reply:
I had the same thought and initially did that, but it's an issue for many tests since they verify the properties of the ByteBuffer and it is not accessible if we do this.

Reply:
> but it's an issue for many tests since they verify the properties of the ByteBuffer and it is not accessible if we do this.

Can it be resolved if we return ```ByteBuffer``` rather than ```Send```?

Reply:
It has to be a `Send` too. I looked at this a bit closer and we can just return `ByteBufferSend` from the static factory method. The thing I attempted before was to combine at the `NetworkSend` level and that doesn't work since it can work with `Send`s that are not backed by a `ByteBuffer`. But it actually works fine at `ByteBufferSend` level. Thanks for the suggestion, please check the update.